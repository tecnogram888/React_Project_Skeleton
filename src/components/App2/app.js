import React from 'react'
import { BrowserRouter as Router, Root, Route } from 'react-router-dom'
import { connect } from 'react-redux'

import Editor from '../Editor'

const test = {
    title: 'titles',
    children: [{
        title: 'sub child',
        children: []
    }, 'asdf', 'next']
}

import css from './../../css/style.css'

class App extends React.Component {
    render() {
        return <Router>
            <div>
                <h1 className="asdf">App2</h1>
                <a>{this.props.text}</a>
                <Editor value={test} />
            </div>
        </Router>
    }
}

const mapStateToProps = (state) => {
    return {
        text: state.text,
    }
}

const wrappedApp = connect(
    mapStateToProps
  )(App)
  

export default wrappedApp
