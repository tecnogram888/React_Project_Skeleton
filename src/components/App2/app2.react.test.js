import React from 'react';
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
import App2 from './app.js';

test('Snapshot Test Practice -- Outline of whole app renders', () => {
    const store = {
        text: 'hello world',
    }
    const mockStore = configureStore()

    const component = renderer.create(
        <Provider store={mockStore(store)}>
            <App2 />
        </Provider>)
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})
