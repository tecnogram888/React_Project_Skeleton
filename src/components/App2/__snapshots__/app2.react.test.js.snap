// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test Practice -- Outline of whole app renders 1`] = `
<div>
  <h1
    className="asdf"
  >
    App2
  </h1>
  <a>
    hello world
  </a>
  <div
    className="editorBox"
  >
    <div>
      Title is titles
    </div>
    <div
      className="editorBox"
    >
      <div>
        Title is sub child
      </div>
    </div>
    <li>
      Value is 
      asdf
    </li>
    <li>
      Value is 
      next
    </li>
  </div>
</div>
`;
